include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -433,182 +422,307 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
         }
     };
 
-    auto skip_tag_name_and_colon = [&it] ()
+    auto skip_until_predicate = [&it] (auto const & predicate)
     {
-        while (!is_char<':'>(*it))
+        while (!predicate(*it))
             ++it;
-        ++it;
     };
 
     auto parse_tag_value = [&] (auto & value) // helper function to parse the next tag value
     {
-        skip_tag_name_and_colon();
+        skip_until_predicate(is_char<':'>);
+        ++it; // skip :
         take_until_predicate(is_char<'\t'> || is_char<'\n'>);
         read_field(string_buffer, value);
     };
 
-    auto parse_until_newline = [&] (auto & value)
+    // Some tags are not parsed individually. Instead, these are simply copied into a std::string.
+    // Multiple tags must be separated by a `\t`, hence we prepend a tab to the string, except the first time.
+    // Alternatively, we could always append a `\t`, but this would have the side effect that we might need to trim a
+    // trailing tab after parsing all tags via `pop_back()`.
+    // Unfortunately, we do not know when we are parsing the last tag (and in this case just not append a tab),
+    // because even though we can check if the line ends in a `\n`, it is not guaranteed that the last tag of the
+    // line is passed to this lambda. For example, the line might end with a tag that is properly parsed, such as `ID`.
+    auto parse_until_tab_or_newline = [&] (std::string & value, std::array<char, 2> raw_tag)
     {
-        take_until_predicate(is_char<'\n'>);
+        take_until_predicate(is_char<'\t'> || is_char<'\n'>);
+        if (!value.empty())
+            value.push_back('\t');
+        value.push_back(raw_tag[0]);
+        value.push_back(raw_tag[1]);
         read_field(string_buffer, value);
     };
 
+    auto print_cerr_of_unspported_tag = [&it] (char const * const header_tag, std::array<char, 2> raw_tag)
+    {
+        std::cerr << "Unsupported SAM header tag in @" << header_tag << ": " << raw_tag[0] << raw_tag[1] << '\n';
+    };
+
     while (it != end && is_char<'@'>(*it))
     {
-        ++it;                                                           // skip @
+        ++it; // skip @
+
+        // char[2] raw_at_tag{};
+        std::array<char, 2> raw_at_tag{};
```

Comment:
`char [2]` did not work, but `std::array<char, 2>` worked.

```bash
.../seqan3/include/seqan3/io/sam_file/detail/format_sam_base.hpp:464:14: error: expected identifier before numeric constant
  464 |         char[2] raw_at_tag{};
      |              ^
```

Reply:
```c++
char raw_at_tag[2]{};
```

is the correct syntax :)

Reply:
but we can keep the std::array